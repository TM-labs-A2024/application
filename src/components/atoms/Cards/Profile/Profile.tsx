import { Link } from '@chakra-ui/next-js'
import { Text } from '@chakra-ui/react'
import { Patient } from '@src/types'
import { formatDistanceToNowStrict } from 'date-fns'
import { es } from 'date-fns/locale/es'
import React from 'react'

export default function Profile({ patient }: { patient: Patient }) {
  return (
    <div className="mb-6 w-full rounded-xl border-2 border-black p-4" data-testid="profile-card">
      <div className="flex flex-col items-center justify-between sm:flex-row">
        <div className="flex flex-col items-start gap-1">
          <Text>
            Nombre: {patient.firstname} {patient.lastname}
          </Text>
          <Text>CI: {patient.govId}</Text>
          <Text>
            Edad:{' '}
            {formatDistanceToNowStrict(new Date(patient.birthdate), {
              locale: es,
              roundingMethod: 'floor'
            })}
          </Text>
        </div>
        <div className="flex w-full flex-col items-end gap-1">
          <Link
            href={`/paciente/${patient.id}`}
            color="white"
            className="rounded-md bg-black px-4 py-2 font-bold text-white no-underline"
          >
            Ver perfil
          </Link>
        </div>
      </div>
    </div>
  )
}
