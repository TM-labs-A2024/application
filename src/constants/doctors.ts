export const doctors = [
  {
    id: 1000000,
    specialities: [1],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 2000000,
    specialities: [2, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 3000000,
    specialities: [3, 5, 7],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 4000000,
    specialities: [4, 1],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 5000000,
    specialities: [5, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 6000000,
    specialities: [6, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 7000000,
    specialities: [7, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 8000000,
    specialities: [8, 9, 5, 1, 3, 4],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 9000000,
    specialities: [9, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 10000000,
    specialities: [10, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 11000000,
    specialities: [11, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 12000000,
    specialities: [12, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 13000000,
    specialities: [13, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 14000000,
    specialities: [14, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 15000000,
    specialities: [15, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 16000000,
    specialities: [16, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 17000000,
    specialities: [17, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 18000000,
    specialities: [18, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 19000000,
    specialities: [19, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 20000000,
    specialities: [20, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 21000000,
    specialities: [21, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 22000000,
    specialities: [22, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 23000000,
    specialities: [23, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 24000000,
    specialities: [24, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 25000000,
    specialities: [25, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 26000000,
    specialities: [26, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 27000000,
    specialities: [27, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 28000000,
    specialities: [28, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 29000000,
    specialities: [29, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 30000000,
    specialities: [30, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 31000000,
    specialities: [31, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 32000000,
    specialities: [32, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 33000000,
    specialities: [33, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 34000000,
    specialities: [34, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 35000000,
    specialities: [35, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 36000000,
    specialities: [36, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 37000000,
    specialities: [37, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 38000000,
    specialities: [38, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 39000000,
    specialities: [39, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 40000000,
    specialities: [40, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 41000000,
    specialities: [41, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 42000000,
    specialities: [42, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 43000000,
    specialities: [43, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 44000000,
    specialities: [44, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 45000000,
    specialities: [45, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 46000000,
    specialities: [46, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 47000000,
    specialities: [47, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 48000000,
    specialities: [48, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  }
]
