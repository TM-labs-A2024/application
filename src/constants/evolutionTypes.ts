export const evolutionTypes = [
  {
    id: '1',
    name: 'Evolución'
  },
  {
    id: '2',
    name: 'Hospitalización'
  },
  {
    id: '3',
    name: 'Alta'
  },
  {
    id: '4',
    name: 'Primera consulta'
  }
]
