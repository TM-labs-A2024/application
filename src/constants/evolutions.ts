export const evolution = {
  specialty: {
    description: '',
    id: '',
    name: ''
  },
  author: '',
  title: '',
  description: '',
  'content-format': '',
  created_at: '',
  payload: {
    type: {
      value: '',
      label: ''
    },
    reason: '',
    description: '',
    history: '',
    examination: '',
    summary: '',
    diagnostic: '',
    comments: ''
  }
}
