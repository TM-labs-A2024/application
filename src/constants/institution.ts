export const institution = {
  id: '3fa85f64-5717-4562-b3fc-2c963f66afa1',
  name: 'Clínica Tex Mex',
  credentials: '14578245',
  address: 'Av. Fernandez Peña, 147.',
  email: 'test@test.com',
  phone: '+582742214441'
}
