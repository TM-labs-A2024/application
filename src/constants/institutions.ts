export const institutions = [
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa1',
    name: 'Clínica Tex Mex 1',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa11',
    name: 'Clínica Tex Mex 2',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa12',
    name: 'Clínica Tex Mex 3',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa13',
    name: 'Clínica Tex Mex 4',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa2',
    name: 'Clínica Tex Mex 5',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa3',
    name: 'Clínica Tex Mex 6',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa4',
    name: 'Clínica Tex Mex 7',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa5',
    name: 'Clínica Tex Mex 8',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
    name: 'Clínica Tex Mex 9',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa7',
    name: 'Clínica Tex Mex 10',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa8',
    name: 'Clínica Tex Mex 11',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa9',
    name: 'Clínica Tex Mex 12',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af10',
    name: 'Clínica Tex Mex 13',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af11',
    name: 'Clínica Tex Mex 14',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af12',
    name: 'Clínica Tex Mex 15',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af13',
    name: 'Clínica Tex Mex 16',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af14',
    name: 'Clínica Tex Mex 17',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af15',
    name: 'Clínica Tex Mex 18',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af16',
    name: 'Clínica Tex Mex 19',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af17',
    name: 'Clínica Tex Mex 20',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  }
]
