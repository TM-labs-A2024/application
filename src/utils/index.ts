import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
