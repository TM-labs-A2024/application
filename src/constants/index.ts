import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
