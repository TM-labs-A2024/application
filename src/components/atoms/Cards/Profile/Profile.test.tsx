import { patient } from '@src/constants'
import { render, screen } from '@test/utils/index'
import { formatDistanceToNowStrict } from 'date-fns'
import { es } from 'date-fns/locale/es'

import '@testing-library/jest-dom'
import Profile from './Profile'

describe('Atoms > Cards > Profile test', () => {
  test('The card shows the name, id and age of the patient', async () => {
    render(<Profile patient={patient} />)

    const textMatcher = (content: string) =>
      content.includes(`${patient.firstname} ${patient.lastname}`)
    const idMatcher = (content: string) => content.includes(patient.govId)
    const ageMatcher = (content: string) =>
      content.includes(
        formatDistanceToNowStrict(new Date(patient.birthdate), {
          locale: es,
          roundingMethod: 'floor'
        })
      )

    const fullNameElement = await screen.findByText(textMatcher)
    const idElement = await screen.findByText(idMatcher)
    const ageElement = await screen.findByText(ageMatcher)

    expect(fullNameElement).toBeInTheDocument()
    expect(idElement).toBeInTheDocument()
    expect(ageElement).toBeInTheDocument()
  })

  test('The url matches the expected', async () => {
    render(<Profile patient={patient} />)

    const enlace = await screen.findByRole('link')
    expect(enlace).toHaveAttribute('href', `/paciente/${patient.id}`)
  })

  test('Matches the snapshot', async () => {
    render(<Profile patient={patient} />)

    expect(await screen.findByTestId('profile-card')).toMatchSnapshot()
  })
})
