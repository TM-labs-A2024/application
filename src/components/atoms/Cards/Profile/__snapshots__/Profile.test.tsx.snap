// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms > Cards > Profile test Matches the snapshot 1`] = `
<div
  class="mb-6 w-full rounded-xl border-2 border-black p-4"
  data-testid="profile-card"
>
  <div
    class="flex flex-col items-center justify-between sm:flex-row"
  >
    <div
      class="flex flex-col items-start gap-1"
    >
      <p
        class="chakra-text css-0"
      >
        Nombre: 
        José Feliciano
         
        Gutierrez García
      </p>
      <p
        class="chakra-text css-0"
      >
        CI: 
        E99885561
      </p>
      <p
        class="chakra-text css-0"
      >
        Edad:
         
        26 años
      </p>
    </div>
    <div
      class="flex w-full flex-col items-end gap-1"
    >
      <a
        class="chakra-link rounded-md bg-black px-4 py-2 font-bold text-white no-underline css-1p1ylbe"
        href="/paciente/3fa85f64-5717-4562-b3fc-2c963f66afa1"
      >
        Ver perfil
      </a>
    </div>
  </div>
</div>
`
